csrc/scalar_value.h

```diff
@@ -163,6 +163,7 @@ inline Pointer operator+(int64_t offset, const Pointer& ptr) {
 
 using ScalarValue = DynamicType<
     Containers<std::vector, Struct>,
+    at::Tensor,
```

Comment:
I'm a little confused here. Is a tensor considered a scalar value?

Reply:
Good point! Tensor is definitely not a scalar value. We needs to have a better name now. Will write a separate PR for renaming.

Reply:
I realized it already has vectors and structs. Is it really like a polymorphic value, isn't it?

Reply:
You got it! I was looking for a word that can correctly describe this. I definitely dislike `EvaluatorValue`, that's why I renamed it to `ScalarValue`. I was just thinking about `DynamicValue`, which I disliked either, because `int i; std::cin >> i;` then `i` is also a dynamic value, it can be `0`, `1`, `2`, `-1`, depending on the user input. `polymorphic` is a good work. I will take that word.